boofuzz/sessions.py

```diff
@@ -96,32 +126,30 @@ def open(self):
         self._fuzz_data_logger.log_info("Connection opened.")
 
     def pedrpc_connect(self):
-        """
-        Pass specified target parameters to the PED-RPC server.
-        """
-        # If the process monitor is alive, set it's options
-        if self.procmon:
-            while 1:
-                if self.procmon.alive():
-                    break
+        warnings.warn(
+            "pedrpc_connect has been renamed to monitors_alive. "
+            "This alias will stop working in a future version of boofuzz.",
+            DeprecationWarning,
+        )
 
-                time.sleep(1)
+        return self.monitors_alive()
 
-            # connection established.
-            for key, value in listitems(self.procmon_options):
-                getattr(self.procmon, "set_{0}".format(key))(value)
+    def monitors_alive(self):
+        """
+        Wait for the monitors to become alive / establish connection to the RPC server.
+        This method is called on every restart of the target and when it's added to a session.
+        After successful probing, a callback is called, passing the monitor.
 
-        # If the network monitor is alive, set it's options
-        if self.netmon:
-            while 1:
-                if self.netmon.alive():
+        :return: None
+        """
+        for monitor in self.monitors:
+            while True:
+                if monitor.alive():
                     break
-
                 time.sleep(1)
 
-            # connection established.
-            for key in list(self.netmon_options):
-                eval('self.netmon.set_%s(self.netmon_options["%s"])' % (key, key))
+            if self.monitor_alive:
+                apply_callback(self.monitor_alive, monitor)
```

Comment:
> The point of apply_callbacks took me a minute to get, and it's only used once here.

I wanted to use it in more places inside callback_monitor.py but stumbled upon the logging that is present there. As I didn't find a quick way to elegantly solve the problem, I ended up not using it.

Because I am still not sure whether this can be reasonably cleaned up, I'd say inline it and remove the function that isn't used anywhere else.

Reply:
And now that I see it, I basically reinvented boofuzz/event_hook.py in a shitty way. Oh well, one argument more to not merge the apply_callback stuff.